src/utils/cellar.ts

```diff
@@ -0,0 +1,238 @@
+import {
+  BigDecimal,
+  BigInt,
+  log,
+} from '@graphprotocol/graph-ts'
+
+import {
+  CellarContract,
+  CellarContract__cellarTickInfoResult,
+} from '../types/Cellar/CellarContract'
+import {
+  Cellar,
+  NFLP,
+  Pool,
+  Token,
+} from '../types/schema'
+import {
+  ONE_BD,
+  ONE_BI,
+  ZERO_BD,
+  ZERO_BI,
+} from './constants'
+
+export function loadCellar(cellarAddress: string): Cellar {
+  log.debug('Invoked loadCellar, address: {}', [cellarAddress])
+  let cellar = Cellar.load(cellarAddress)
+  if (cellar == null) {
+    throw new Error('Could not find Cellar:'.concat(cellarAddress))
+  }
+
+  return cellar as Cellar
+}
+
+export function initCellar(contract: CellarContract, cellarAddress: string): Cellar {
+  log.debug('Invoked initCellar', [])
+  let cellar = new Cellar(cellarAddress)
+  cellar.token0 = contract.token0().toHexString()
+  cellar.token1 = contract.token1().toHexString()
+
+  // TODO: derive this from token and fee
+  // PORT: https://github.com/Uniswap/uniswap-v3-sdk/blob/main/src/utils/computePoolAddress.ts
+  cellar.pool = '0x8ad599c3a0ff1de082011efddc58f1908eb6e6d8' // USDC-ETH 0.3%
+  cellar.feeTier = BigInt.fromI32(contract.feeLevel())
+  cellar.managementFee = BigInt.fromI32(contract.fee())
+
+  cellar.totalDepositAmount0 = ZERO_BD
+  cellar.totalDepositAmount1 = ZERO_BD
+  cellar.totalDepositUSD = ZERO_BD
+  cellar.totalValueLockedToken0 = ZERO_BD
+  cellar.totalValueLockedToken1 = ZERO_BD
+  cellar.totalValueLockedUSD = ZERO_BD
+  cellar.feesCollectedToken0 = ZERO_BD
+  cellar.feesCollectedToken1 = ZERO_BD
+  cellar.feesCollectedUSD = ZERO_BD
+  cellar.feesReinvestedToken0 = ZERO_BD
+  cellar.feesReinvestedToken1 = ZERO_BD
+  cellar.feesReinvestedUSD = ZERO_BD
+
+  return cellar
+}
+export function getCellarTickInfo(contract: CellarContract): CellarContract__cellarTickInfoResult[] {
+  log.debug('Invoked getCellarTickInfo', [])
+  let result = new Array<CellarContract__cellarTickInfoResult>()
+
+  // TODO: update contract with fn that returns the full array
+  let i = ZERO_BI
+  let isOutOfBounds = false
+  while (isOutOfBounds == false) {
+    let tickResult = contract.try_cellarTickInfo(i)
+
+    if (tickResult.reverted) {
+      log.debug('getCellarTickInfo reverted', [])
+      isOutOfBounds = true
+    } else {
+      log.debug('the tickResult.value {}', [tickResult.value.value0.toString()])
+      result.push(tickResult.value)
+      i = i.plus(ONE_BI)
```

Comment:
Oh man, this is what we have to do to find the ticks initialized on a cellar? This is pretty bad - these loops can take over a minute.

If there is no alternative, instead of incrementing by 1, you should be able to increment by `cellar.pool.tickSpacing`

Reply:
@kkennis I forgot to expose a function to return the full array. We should probably do that and also fix the rebalance / reinvest events to send the full fees collected instead of the maintenance fee.